File: brzy-scheduler/src/main/java/org/brzy/mod/scheduler/Cron.java
package org.brzy.mod.scheduler;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

/**
 * Document Me..
 *
 * @author Michael Fortin
 */
@Retention(RetentionPolicy.RUNTIME)
public @interface Cron {
    String value() default "* * * * *";// every second
    String method() default "execute";
}

File: brzy-scheduler/src/main/scala/org/brzy/mod/scheduler/Cron.scala
package org.brzy.mod.scheduler

import org.brzy.service.Service

/**
 * Services that wish to be executed on a cron like schedule should implement this trait.
 * 
 * @author Michael Fortin
 */
trait Cron extends Service {

  /**
   * The cron expression used to calculate the next execution.
   */
  val expression = "0/5 * * * * ?"

  /**
   * the function called to execute the cron job.
   */
  def execute()
}